# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/VipS-2004/flyrank1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane and Why

I chose **Refresh / Content Opportunity Scoring** because it focuses on identifying pages that should be reviewed first for content improvement. This aligns with the internship objective of helping prioritize limited review time using observable search and engagement signals. My goal is to build a transparent ranking system that recommends which pages deserve attention first.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## The Question

**Research Question**

Which content pages should be reviewed first for refresh or improvement based on observable search and engagement signals?

**Decision**

Decide which pages should receive review priority.

**Action**

Review, update, expand, or monitor the selected pages.

**Cost of a Wrong Recommendation**

A false positive wastes time reviewing pages that do not need improvement.

A false negative allows important pages to continue losing visibility and traffic.

**Why ML?**

Machine learning can combine multiple search and engagement signals to rank content review opportunities more effectively than simple rule-based methods.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

## Quick Look at the Data

I explored the starter dataset to understand whether it contains enough information to support the Refresh / Content Opportunity Scoring lane.

The code below loads the dataset and reports a few basic statistics such as the number of pages, the distribution of trend directions, and engagement rate statistics. These numbers help justify why this lane is worth exploring over the next seven weeks.

In [10]:
import os

print(os.getcwd())
print(os.listdir())

/content
['.config', 'flyrank1', 'sample_data']


In [11]:
!git clone https://github.com/VipS-2004/flyrank1.git

fatal: destination path 'flyrank1' already exists and is not an empty directory.


In [12]:
!find /content/flyrank1 -name "*.csv"

/content/flyrank1/outputs/refresh_queue_sample.csv
/content/flyrank1/data/raw/content_refresh_anonymized.csv


In [13]:
import pandas as pd

df = pd.read_csv("/content/flyrank1/data/raw/content_refresh_anonymized.csv")

print("Dataset Shape:", df.shape)

print("\nTrend Direction Counts:")
print(df["trend_direction"].value_counts())

print("\nEngagement Rate Statistics:")
print(df["engagement_rate"].describe())


Dataset Shape: (30000, 44)

Trend Direction Counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Engagement Rate Statistics:
count    30000.000000
mean         2.534520
std          8.310096
min          0.000000
25%          0.000000
50%          0.000000
75%          1.350000
max        100.000000
Name: engagement_rate, dtype: float64


## Observations from the Starter Dataset

The starter dataset contains **30,000 content pages** and **44 features**, providing enough data to explore content refresh opportunities.

The trend direction distribution shows that **16,262 pages are declining**, **5,962 are stable**, **4,388 are improving**, **2,236 are new**, and **1,152 are flat**. This suggests that a significant portion of the content may require review.

The average engagement rate is **2.53%**, with a median of **0%**, indicating that many pages receive very little user engagement. This further supports the need for a data-driven approach to prioritize which pages should be reviewed first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This project is intended to support decision-making by identifying pages that may benefit from content review.

The analysis can identify patterns and relationships in the available data, but it cannot prove that refreshing a page will improve its performance.

The recommendations should assist human reviewers rather than replace their judgment.

## Self Check

-  I selected one predefined lane.
-  I explained why I chose this lane.
-  I identified the decision, action, and cost of a wrong recommendation.
-  I included supporting statistics from the starter dataset.
-  I explained what my work can and cannot claim.